### PR TITLE
Add 'podman restart' command

### PR DESCRIPTION
**Pull request by mheon**

### CODE REVIEW

File: docs/podman-restart.1.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+## NAME
+podman restart - Restart a container
+
+## SYNOPSIS
+**podman attach [OPTIONS] CONTAINER [CONTAINER...]**
+
+## DESCRIPTION
+The restart command allows containers to be restarted using their ID or name.
+Containers will be stopped if they are running and then restarted.
```
Comment (baude) [inline docs/podman-restart.1.md]: This implies if a container is stopped, restart will just start it? If so, probably worth saying.

**Comment by baude** [comment]
e2e tests?

**Comment by mheon** [comment]
@baude Oops, lemme add those

**Comment by mheon** [comment]
Added E2E tests

File: test/e2e/restart_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		session2.WaitWithDefaultTimeout()
+		Expect(session2.ExitCode()).To(Equal(0))
+	})
+
+	It("Podman restart running container", func() {
+		_ = podmanTest.RunTopContainer("test1")
+		ok := WaitForContainer(&podmanTest)
+		Expect(ok).To(BeTrue())
+
+		session := podmanTest.Podman([]string{"restart", "test1"})
```
Comment (baude) [inline test/e2e/restart_test.go]: nit: how about using -l here instead of a name given that in the next test you use names

**Comment by baude** [comment]
one nit, else lgtm

**Comment by baude** [comment]
LGTM

File: cmd/podman/restart.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+	runtime, err := getRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "error creating libpod runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	var lastError error
+
+	timeout := c.Uint("timeout")
```
Comment (rhatdan) [inline cmd/podman/restart.go]: Can't we just set timeout to -1, and then allow user to specify the timeout. No need for special boolean.   Or just set it to 0, and make users specify timeout > 0.

File: completions/bash/podman [hunk 4]
```diff
@@ -1313,6 +1313,16 @@ _podman_run() {
 	_podman_container_run
 }
 
+_podman_restart() {
```
Comment (rhatdan) [inline completions/bash/podman]: You need podman defined below.

**Comment by rhatdan** [comment]
We also need commands.md updated.

File: cmd/podman/restart.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	for _, id := range args {
+		ctr, err := runtime.LookupContainer(id)
+		if err != nil {
+			if lastError != nil {
+				fmt.Fprintln(os.Stderr, lastError)
+			}
+			lastError = errors.Wrapf(err, "unable to find container %s", id)
+			continue
+		}
+
```
Comment (rhatdan) [inline cmd/podman/restart.go]: How about 
if isset(timeout) 
timeout = timeout
else
timeout = ctr.StopTimeout()
fi
restrartCtr(ctr, timeout)

File: commands.md [hunk 6]
```diff
@@ -29,6 +29,7 @@
 | [podman-ps(1)](/docs/podman-ps.1.md)                     | Prints out information about containers                                   |[![...](/docs/play.png)](https://asciinema.org/a/bbT41kac6CwZ5giESmZLIaTLR)|
 | [podman-pull(1)](/docs/podman-pull.1.md)                 | Pull an image from a registry                                             |[![...](/docs/play.png)](https://asciinema.org/a/lr4zfoynHJOUNu1KaXa1dwG2X)|
 | [podman-push(1)](/docs/podman-push.1.md)                 | Push an image to a specified destination                                  |[![...](/docs/play.png)](https://asciinema.org/a/133276)|
+| [podman-restart](/docs/podman-restart.1.md)              | Restarts one or more containers                                           ||
```
Comment (TomSweeneyRedHat) [inline commands.md]: We should add this line to podman.1.md too.

File: docs/podman-restart.1.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+**podman attach [OPTIONS] CONTAINER [CONTAINER...]**
+
+## DESCRIPTION
+The restart command allows containers to be restarted using their ID or name.
+Containers will be stopped if they are running and then restarted. Stopped
+containers will not be stopped and will only be started.
+
+## OPTIONS
+**--timeout**
+Timeout to wait before forcibly stopping the container
```
Comment (TomSweeneyRedHat) [inline docs/podman-restart.1.md]: Should we note the default timeout here?

Comment (TomSweeneyRedHat) [reply docs/podman-restart.1.md]: Actually do we have one, or only just in the tests?  I'm not seeing it in libpod.conf. If we don't have one, ignore this.

Comment (mheon) [reply docs/podman-restart.1.md]: It's hardcoded somewhere. Probably a good idea to make it configurable for the entire runtime at some point...

The timeout isn't in the `podman stop` manpage so I omitted it here

File: cmd/podman/restart.go (hunk 5 above)
Comment (mheon) [reply cmd/podman/restart.go]: Fixed

File: commands.md (hunk 6 above)
Comment (mheon) [reply commands.md]: Fixed

**Comment by mheon** [comment]
I think this is ready

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @baude @umohnani8 PTAL

**Comment by baude** [comment]
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by TomSweeneyRedHat** [comment]
LGTM, it would be nice to add the time out period to the conf at some point, but not needed for this PR.

**Pull request closed without merging**